### ISSUE TITLE
Infinite loop reading configuration files

### ISSUE THREAD

**Issue by HiTEC** [config-r/config-r#190]
Given the following CSX configuration file:
```c#
Add("Test", 3);
while (true) ;
```
My app hangs at `ConfigR.Config.Global`. According to this [SO question](http://stackoverflow.com/q/15982013/5869), it doesn't appear there is a simple solution.

**Comment by adamralph** [comment]
Yes, when using ConfigR, your app executes the C# in the config file synchronously which means if that the config code hangs then your app hangs.

Is this causing you an issue when things take 'too long' to happen in your config file or is this just an academic enquiry? If the former, we could probably introduce an execution timeout somehow, perhaps even with a cancellation token. Would that solve your problem?

**Comment by HiTEC** [comment]
I was evaluating the library, and the expectation was a configuration timeout exception. I don't want to introduce configuration lookup calls that could not return in certain cases. Ideally I would want my client to only expect two cases when talking to configuration:

- success, the configured value
- failure, a `ConfigurationException` that could be a syntax error, runtime exception, or timeout

In the case of timeout, I would want that configurable, so over time I could measure & expect configuration calls to return as they have in the past.

Thanks for the quick response!

**Comment by adamralph** [comment]
@HiTEC that sounds like a reasonable design. This will need some spiking around the script execution to see how it could work.
